Custom skulls

Preloading opened the pull request.

WesleyVanNeck commented:
@Loganh4005 hey try and use this it makes merging more easyer
https://www.gitkraken.com/git-client

Camotoy committed:
Add WORLD_IMMUTABLE flag to spectator mode (#1034)

Prevents the client from trying to interact with blocks

Camotoy committed:
Hide stack trace if an SRV record is unable to be found. (#1046)

Camotoy committed:
Update submodules (#1047)

Camotoy committed:
Update to latest master

OnlyBMan committed:
Merge pull request #7 from bundabrg/feature/custom-skulls-pr

Feature/custom skulls pr

Camotoy committed:
Update to 1.16.2-pre1

rtm516 committed:
Fix duplicate info and add more info to dumps (#1058)

* Fix duplicate info and add more info to dumps

* Add gui to standalone dump info

rtm516 committed:
Check if the passenger is null before trying to update metadata (#1001)

* Check if the passenger is null before trying to update metadata

* Fix variable name

rtm516 committed:
Move MEGABYTE to a constant (#1059)

rtm516 committed:
Add translation badge and remove manual remapping of language codes (#1060)

toinouH committed:
clone-remote-port option Updated (#1061)

* Added clone remote port option for bukkit, bungee and velocity

* Added clone remote port option for sponge

* Changed clone-remote-port description in config.yml

* Update config.yml

Updated config.yml to include a better description of the clone-remote-port option

* Updated GeyserSpongePlugin

An incorrect port was being edited before (remote instead of bedrock)

* Update config.yml

Co-authored-by: TeaNoDonuts <blackalegator@gmail.com>

rtm516 committed:
Update closest color conversion (#1057)

ViaVersion altered their color conversion to fix an issue and this just copies those changes

rtm516 committed:
Clean copyright message and update all files (#1053)

bundabrg committed:
Fix Anvil renames by trying a component first then fallback to plain text (#1052)

Closes #1039

bundabrg committed:
Test if slot 50 used and under what conditions. (#1028)

Camotoy committed:
Add rabbit jumping animation (#1027)

* Add rabbit jumping animation

This isn't perfect as Bedrock uses a duration and Java just sends the jumping animation. There may be something else missing from the puzzle piece.

* Remove debug line

Camotoy committed:
Only send metadata update once per Java metadata packet (#1022)

While this doesn't fix any bugs, it may be a slight performance enhancement as we aren't sending multiple packets per one Java entity metadata packet.

Camotoy committed:
Support immediate respawn gamerule (#970)

* Support immediate respawn gamerule

This commit now supports immediate respawn if the server enables it - both on the setting being applied on join and the setting being modified in-game. This also refactors the respawning process to more closely match BDS behavior - nothing broke in my testing but more testing is needed.

* Reuse spawned variable instead of creating new variable

rtm516 committed:
Fix clone-remote-port (#1062)

Arktisfox committed:
Small entity metadata fix, other player bow implementation (#685)

* Move blocking case to LivingEntity, and make other players bows animate.

This moves metadata ID 7 to LivingEntity, it's proper place. It also sets the 'USING_ITEM' flag which animates other players bows.

* Add skeleton aiming support

Skeletons don't have support of pushing their bows back on Bedrock, but this allows them to hold their arms up

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Redned235 committed:
Add test server to README

Camotoy committed:
Add customizable MTU support (#1068)

* Add customizable MTU support

Fixes clients being unable to connect in rare instances.

* Make config.yml nicer

Camotoy committed:
Check for display tag when translating anvil contents (#1073)

AJ-Ferguson committed:
Remove Y pos workaround in BedrockItemFrameDropItemTranslator (#1037)

Camotoy committed:
JavaEntityMetadataTranslator: replace stack trace with concise warning (#1086)

* JavaEntityMetadataTranslator: replace stack trace with concise warning

Removes the stack trace given when a ClassCastException occurs and replaces it with a friendlier message. Class cast errors will happen since some servers send incorrect values, and apparently it is default Minecraft behavior to ignore them.

* Update languages submodule

AJ-Ferguson committed:
Update effects mappings (#949)

* Update effects mappings

* Use STOP_RECORD as the default record instead of null

* Add comments

* Update mappings submodule

* Update MCProtocolLib and effects

* Change level event used for EVAPORATE effect

The bedrock client plays an additional sound when using CAULDRON_EXPLODE.
The java client does not play any sound.

* Update mappings submodule

Camotoy committed:
Merge latest master

rtm516 committed:
Update mappings submodule to fix 1.16 slabs and stonecutter (#1089)

Camotoy committed:
Update to 1.16.2-pre2

bundabrg committed:
Fix Skin Caching and Fix Skin Restorer (#680)

* Fix Skin Caching

Changes:
* Instead of caching a skin based upon the player we cached it based upon the textureURL. This means multiple players with the same skin will benefit from the cache and more importantly will mean a player changing their skin will not get a false cache hit.
* This should fix all issues with SkinRestorer and will now correctly show the skin both to the player themselves and to other players

Closes #518

* Remove duplicated code

* Minimize playerlist updates

Changes:
* All async skin stuff will now just update skins and not be involved with sending the session to the player. This eliminates issues where the player list changes whilst an async task is occuring plus it means no invisible players while retrieving skin.
* Fix bug when retrieving cached skin

* When sending PlayerList packets ensure the skins have appropriate skinIds so the Bedrock client will cache hit/miss as needed

* Make sure to add and remove player when setting skin if they do not belong on the playerlist

* Make use of AuthData UUID when removing the player

* Revert removal of checking if entity is valid when initialized

This section is supposed to send all spawned entities in the java world to a player only after they've initialized. By removing this check it would also be sending entities that exist but are not spawned.

* Optimizations

Changes:
* Check for duplicate requests based on textureURL instead of player ID
* Don't use the PlayerSkinPacket. It duplicates the data sent in the PlayerListPacket and without it the players still get skin updates.

* Support caching of skins to disk based on configuration variable

If a skin is downloaded it will be saved to `cache/skins` using a base64 encoded filename of the textureUrl, if allowed by setting a non 0 value for the configuration variable `cache-skins`

When reading a skin we try load it from a cache file first before trying to download it.

We don't yet expire them but do update their last modification so we know which ones have been accessed.

* Update `config.yml` with cache-skins directive, defaulting to disabled

* Merge Fixes

* Cache all images instead of just skins

Changes:
* Move the image caching from skins to where images may get downloaded so this also covers capes and anything else that uses the same method of image retrieval
* Updated config value from `cache-skins` to `cache-images`
* Updated cache location from `cache/skins` to `cache/images`
* Images are stored in png format with a uuid. This may make debugging easier as they can be directly opened.

* Implement cached image expiry

If `cache-images` is set to a value greater than 0 then a scheduled task will occur once a day that will remove images with a modification date older than the value in days.

* Force skin changes as trusted

* Resolve PR queries

* Fix signed int causing issues calculating expiry time for images

* Reset Defaults to 0 and implement Google Timed Eviction cache for Images

* Add memory cache for Capes

Co-authored-by: Brendan Grieve <brendan.grieve@zepli.com.au>
Co-authored-by: bundabrg <bundabrg@grieve.com.au>

Camotoy committed:
Change scoreboard errors to debug only (#781)

Prevents errors from occuring that don't stop operation of Geyser.

Camotoy committed:
Implement proper mappings for pistons, dropper, dispenser (#1103)

This commit gets rid of the hacky workaround implemented for pistons, droppers and dispensers and actually implements the vanilla data values.

Camotoy committed:
GeyserSession: always send naturalRegeneration=false gamerule (#1097)

This essentially gives the server full control over the health visual.

rtm516 committed:
Add support for client side settings (#1035)

* Port code from #486

Co-authored-by: Luke <32024335+lukeeey@users.noreply.github.com>

* Fix and clean code and add default gamemode changing

* Clean copyright

* Remove direct modification of server, clean up code and add player list xuid fetching.

* Move to custom settings menu

* Move sendAdventureSettings to GeyserSession

* Add javadoc comments

* Add translation support

* Remove updated copyright

* Clean up

* Clarify some javadoc comments

* Remove obsolete code

* Update languages submodule

* Fix javadoc comments

* Fix compile

Co-authored-by: Luke <32024335+lukeeey@users.noreply.github.com>
Co-authored-by: Redned <redned235@gmail.com>

Redned235 committed:
Fix javadoc errors

Commit:
Fix fire not extinguishing on server side when on bedrock

Fixes #875
Fixes #906

Camotoy committed:
Update to 1.16.2-rc1

rtm516 committed:
Fix Shulker color and open state (#1113)

Camotoy committed:
Update submodules (#1109)

rtm516 committed:
Fix dust particles type (#1108)

Camotoy committed:
Remove debug line

rtm516 committed:
Fix quotes breaking formatted message strings (#1118)

EasyClifton committed:
Add Command Block and Structure block to the fix list (#1119)

* Add Command Block and Structure block to the fix list

* Add Horse Inventory to the fix list

Added Horse to the What's left to be added/fixed list.

Camotoy committed:
Update for protocol v409

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into feature/1.16.2

Camotoy committed:
Update mappings

Camotoy committed:
Update to 1.16.2-rc2; add multiversion support

Camotoy committed:
Add v408 as the default protocol

smh mojang

Camotoy committed:
Update to 1.16.2

rtm516 committed:
Fix chat translation parameters not having color sometimes

rtm516 committed:
Change version number to 1.1.0

Camotoy committed:
Bump MCProtocolLib

Redned235 committed:
Merge pull request #1127 from GeyserMC/feature/1.16.2

1.16.2 Support

bundabrg committed:
Fix Skin Selfie (#1131)

Co-authored-by: bundabrg <bundabrg@grieve.com.au>

Camotoy committed:
EntityUtils: Properly map 1.14 entity status effects (#1133)

Previously, Hero of the Village and Bad Omen effects were mapped to 0. This commit updates them to their proper Bedrock values.

Fixes #1129

bundabrg committed:
Fix Alex/Steve skins being sent incorrectly (#1135)

* Return permanent skins (alex/steve) when queried instead of returning an empty skin due to invalid lookup
* Fix Alex/Steve being shown incorrectly due to java signed integers

Co-authored-by: bundabrg <bundabrg@grieve.com.au>

Redned235 committed:
Kick player with invalid chain data for additional security

The client should disallow players to join servers if they're not logged in, however this just adds a second layer of security in the event that it's somehow bypassed.

Camotoy committed:
LoginEncryptionUtils: Add proper string for no Xbox account (#1139)

Camotoy committed:
Update to latest MCProtocolLib (#1140)

Camotoy committed:
ConnectorServerEventHandler: set default packet handler (#1148)

This allows disconnect packets to be sent and kick unsupported versions of the game.

rtm516 committed:
Add support for sensitive data in dumps (#1149)

* Add sensitive dumps

* Add better arg handling and offline dumps

* Add sensitive parameters for plugin IPs

* Add sensitive property to the Bedrock remote address

Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Camotoy committed:
Translate RAIN_STRENGTH to Bedrock client (#1151)

* Translate RAIN_STRENGTH to Bedrock client

Previously Geyser ignored RAIN_STRENGTH and instead relied on START_RAIN and STOP_RAIN only. This is unreliable on a vanilla server as these values are swapped around. This commit also implements thunder strength which was untranslated.

* Update rain code in JavaRespawnTranslator

R-Josef committed:
Follows specified address/port in remote config for plugin versions: fix #1110 (#1145)

* fix #1110

* updating comments in config.yml

* Fix indentation

* Centralize localhost retrieval; remove unnecessary Docker check

* Add config.yml

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
FireworkEntity: don't process if item is null (#1160)

Camotoy committed:
Translate specific messages (#1161)

Camotoy committed:
Fix Xbox authentication and add support for proxies (#1162)

Waterdog and ProxyPass will work when `enable-proxy-connections` is set to true at the expense of security.

Redned235 committed:
Disable fireworks for consoles (Addresses #1083, #1164)

Not ideal, but there isn't a whole lot we can do as this is a game bug within console versions.

Redned235 committed:
Set player motion when explosion takes place

Redned235 committed:
Only one of the values here needs to be greater than 0

jcxldn committed:
Add 'passthrough-protocol-name' config option (#1124)

* Initial version (tested)

* Don't bump config version

* Misc changes

* Add punctuation to config

Camotoy committed:
Auto-configure more if setting is enabled (#1168)

* Auto-configure more if setting is enabled

- Geyser dumps now show if the config was automatic
- Floodgate is now automatically detected if the address is also automatically found
- If the plugin versions' servers have the listening address set to something different, set our remote address to that

* Fix Sponge config

* Remove redundant Getter

Camotoy committed:
Implement an enchantment table GUI (#1177)

Until 1.16, enchantment tables were impossible to implement properly in Geyser. When a user selects an enchantment in Bedrock, the client creates the book on its end and assumes the server is OK with it. Java requires a button to be pressed to select the enchantment. With 1.16, server authoritative inventories remove that on Bedrock. However, until our inventory rewrite is finished we are still stuck without enchantment table support. This commit serves as an alternative as we wait.

Enchantment table GUI support is still impossible since we are using the pre-1.16 inventory system. To solve this, this commit replaces the enchantment table GUI with a hopper GUI. The first slot serves as the spot you place the weapon. The second slot acts as the lapis slot - Geyser prevents any item from going in there that is not lapis. The final three slots act as the buttons; an enchanted book acts as each button, with the ability to show the translated text of each enchantment.

https://cdn.discordapp.com/attachments/613194828359925800/746164042359504927/unknown.png

Camotoy committed:
Don't play a sound when falling onto a block (#1175)

Camotoy committed:
Prevent a comma from appearing if Geyser fails to bind (#1174)

Camotoy committed:
Fix NPE if block data string isn't in mappings (#1172)

This happens when the server version is below the current version and the block state changed. A better solution would be to use ViaVersion to translate the block state strings but this would require getting the server version and figuring out mappings from there.

jord1e committed:
Add visual support for signs colored with dye (#1180)

* Fix dyed signs in Bedrock Edition

Add visual support (in Bedrock Edition) for signs colored with dye (in Java Edition)

* Javadoc for getBedrockSignColor(string)

* Simplified getBedrockSignColor(string)

EasyClifton committed:
Add Loom to fix list (#1185)

Added loom to the What's Left to be Added/Fixed list.

Comstepr committed:
Add smithing table to what's left to be added/fixed (#1187)

Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Saraistupid committed:
Add a warning at start about movement translation (#1069)

Co-authored-by: Redned <redned235@gmail.com>
Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Camotoy committed:
Implement (hopefully) temporary dimension switching fix (#1188)

This fixes rare (?) instances where dimension switching doesn't finish loading on the client. Ideally a proper fix would send the finishing packets in the correct order but I didn't get far in this regard.

Fixes #1154 and #1072.

Other miscellaeous chunk-related fixes have also been included here.

Camotoy committed:
BedrockEmoteTranslator: ensure sending player is valid for all other sessions (#1194)

Camotoy committed:
Update mappings (#1196)

Camotoy committed:
Add shulker box item tooltip translating (#1189)

* Add shulker box item tooltip translating

This commit adds support for previewing the items inside of a shulker box. This does not do a full translation, and only does enough to translate the item information to the client, so as to prevent any accidental item modifying/removing on creative mode.

* Swap values

Camotoy committed:
Translate client-computed recipes (#1181)

* Translate client-computed recipes

A handful of recipes are complex enough on Java Edition that the client simply calculates them after getting an assurance that they are valid recipes. This PR stores those recipes in a Bedrock-compatible format in mappings, then generates the CraftingData information on startup to send to the Bedrock client when called. This fixes firework rocket and star crafting, and fixes leather armor and shulker box dyeing.

The recipe information for everything except leather armor was taken right from the Bedrock server. The leather armor had to be created separately (see https://github.com/DoctorMacc/LeatherDyeingCreation). There will be a slight visual difference in the crafting result preview if the armor is not perfectly dyed to one of the sixteen colors, but this is a visual issue that will persist unless we calculate every single possbile combination.

* Revert other changes

* Register shulker box recipes properly

* Add break

* Update mappings

jcxldn committed:
Add Server Name config option (#1170)

* Add bedrock.server-name config option

* Fix spelling mistake oops!

* Remove trailing whitespace

abeshi-softwire committed:
Adding option to use different config file for standalone (#1102)

* Added config option to standalone Geyser

* Cleanup

* Added --gui, --nogui options

* Made new options read default config.yml from correct place internally

* Changed to locale strings rather than hardcoded English

* Using separate options texts

* Changed '-c' to be string parameter so it isn't translated

rtm516 committed:
Fix reflections relocation (#1199)

Tim203 committed:
Scoreboard improvements (#1166)

* Added a way to check if debug logging is enabled

* Improved scoreboard performance

* Include Teams in pps and return pending pps instead when higher then pps

Some servers have a huge amount of score packets when the player logs in, but before this commit, only after the first high pps (packets per second) the ScoreboardUpdater will be used (after pending packets per second have been moved to packets per second). But this commit fixes that the ScoreboardUpdater can be used on the second that the pps is getting high.

* Fixed team pre + suffix "null" issue and added threshold config option

Fixed team pre + suffix "null" issue.
When the prefix and/or suffix of a Team is null, "null" will be returned instead of null (Due to the way that MCProtocolLib is made and designed). This is fixed by simply checking if the prefix and/or suffix equal "null" and if that is the case, replace it with "".

Added threshold option.
Gave the person who is running Geyser an option to specify the first Scoreboard packets per second threshold to further improve performance by lowering the setting or decrease performance by relaxing the setting a bit. The value can't be higher then 250 (the second threshold), because it'll always choose the lowest threshold.

* Forgot to bump config version

* Small changes

* Reverted version bump, changed Sponge config, changed FloodgateKeyLoader

Reverted version bump
Camotoy said that you only need to bump the config version if the change is breaking, the config version bump has been reverted.

Changed Sponge config
The Sponge config has been modified to look like the other platform configurations.

Changed FloodgateKeyLoader

* Changed default-locale and (remote) address as requested by Camotoy

* Reduce bandwidth and a few final tweaks

* Made the scoreboard-packet-threshold a bit higher due to improvements

rtm516 committed:
Tweaks to support Android (#1206)

* Downgrade reflections to 0.9.11

* Add comment explaining downgrade

* Move to pre-build reflections

* Update skins to use https and relative cache dir

* Move to https OptiFine cape url

* Add javadoc to isProduction

* Add ANDROID as a platform type

* Re-ordered PlatformType

* Change stop command to call onDisable

Camotoy committed:
connector/pom.xml: Move back to main MCProtocolLib repository (#1211)

Fixes a regression where ClientPlayerAbilitiesPacket was sending the incorrect flag.

jackson-57 committed:
Fix minor spelling error in issue template (#1214)

Redned235 committed:
Show the supported Bedrock versions in the version command rather than just the default codec

Camotoy committed:
ShulkerBoxItemTranslator: Ensure Items ListTag is present (#1221)

* ShulkerBoxItemTranslator: Ensure Items ListTag is present

* Compile

Camotoy committed:
ScoreboardUpdater: Quick fix to lessen CPU usage (#1238)

This prevents one/multiple CPU cores from taking up 100% usage. A better, permanent fix will replace this in the coming days.

Camotoy committed:
LocaleUtils: don't NPE if no default locale mapping exists (#1239)

Camotoy committed:
JaveNotifyClientTranslator: Translate invalid bed message (#1212)

This isn't sent as its own message but as a specific event.

Camotoy committed:
GeyserSession: Always set Keep Inventory to true (#1213)

* GeyserSession: Always set Keep Inventory to true

This prevents the client from removing items on death in creative mode if Keep Inventory is true, but doesn't break existing behavior. Essentially, this assures full server-side behavior of the inventory during death.

* Small comment update

* OK, it was fine before the last commit, but make it better

Comstepr committed:
Minor changes to config.yml (#1224)

Camotoy committed:
Relocate the rest of our dependencies (#1227)

- Relocate all of our dependencies. This does not include MCProtocolLib and Nukkit dependencies at this time as there are no other known plugins that use these dependencies.
- Switch to a static commit for Adventure dependencies.

Tested working on all versions.

Camotoy committed:
Implement correct sign wrapping (#1228)

* Implement correct sign wrapping

This commit ensures that the auto-wrapping nature of Bedrock with signs is corrected. If a Bedrock player sends a sign that is auto-wrapped, it will now be interpreted by Geyser to fit on multiple lines. Additionally, Geyser will crop incoming sign text to prevent auto-wrapping.

* Don't wrap if it's the last line

Camotoy committed:
Parrot mounting fixes (#1236)

* Parrot mounting fixes

- Fix duplicate parrots when a parrot leaves the player
- Fix rotation of parrots

* Remove critical debug information

Camotoy committed:
Translate scoreboard nametag visibility (#1240)

This commit adds support for name tag visibility in teams. If a player is set to hide their nametag, it will be hidden from the Bedrock client. Notably, this fixes most NPC nametag hiding, including Citizens. This does not fix some NPC nametag hiding - there are several NPCs in Hypixel that still have a nametag show up, and they are not a part of any team.

Camotoy committed:
Relocate Google Common (#1242)

Fixes NoSuchMethodErrors from occuring on certain Spigot servers.

Camotoy committed:
Non-full-chunk support (#574)

This commit adds non-full chunk support if chunk caching is enabled.

Camotoy committed:
Update mappings to fix 1.16 wood slabs (#1249)

Camotoy committed:
GeyserSession: send command permission level OPERATOR if qualified (#1254)

Mobile clients have a GUI for commands that shows if CommandPermission.OPERATOR or higher is sent. The commands present all require OP permission 2 or higher; therefore we set that command permission if the server tells us we have a OP permission level of 2 or higher.

Camotoy committed:
Update to 1.16.3 (#1272)

* Update for 1.16.3-rc1

* Update to 1.16.3

* Update README

* Update MCProtocolLib

Camotoy committed:
BedrockInventoryTransactionTranslator: check to make sure bucket usage is on purpose (#1280)

Otherwise buckets can be activated when opening block inventories.

Camotoy committed:
Check if Fireworks tag is null (#1255)

Thank you Mineplex, very cool.

Camotoy committed:
Translate CanPlaceOn/CanDestroy NBT (#1253)

* Translate CanPlaceOn/CanDestroy NBT

This commit adds support for the translation of the CanPlaceOn/CanDestroy NBT for Bedrock clients.

* Remove debug line

Camotoy committed:
PlayerEntity: despawn even if still on the player list (#1263)

Fixes LibsDisguises not working, as it uses the same entity ID for the disguised entity and player. The player still appears on the player list.

Camotoy committed:
DumpInfo: Mark internal IP as sensitive (#1264)

Sometimes the internal IP is the external IP of the server.

Camotoy committed:
Pick block improvements (#1265)

* Pick block improvements

- Creative block picking is now implemented. If the survival-styled block picking fails, then the item is created, following Java-style mechanics.
- Entity 'picking' is also implemented. The item is crafted using the same mechanics, and the same rules apply as normal block-picking (except it only works in creative mode, following Java.

* Switch some logic around

Camotoy committed:
JavaCollectItemTranslator: check null for entities (#1267)

Camotoy committed:
Disconnect player if Java sends disconnection (#1274)

Camotoy committed:
Implement command block and jigsaw support (#1291)

* Implement command block and jigsaw support

- Command block UI is now fully implemented to match Java Edition.
- Command block minecarts are now supported.
- Command blocks now show the correct type of command block.
- Jigsaw blocks are translated.

Structure blocks can be implemented, but these will be trickier as there are significant GUI differences between Java and Bedrock.

* Add more detail about command block minecart color

* Set PlayerPermission.OPERATOR to allow command blocks to be destroyed

Camotoy committed:
Remove Protocol v409 support (#1300)

* Remove Protocol v409 support

Protocol has dropped support for this version.

* Fix movement

* Use a static commit for Protocol

Camotoy committed:
Fix inconsistencies with players and the player list (#1298)

* Fix inconsistencies with players and the player list

This commit makes the player list entry packet control the player cache, fixing inconsistencies that appeared when removing the override on despawning the player.

* Update comments

rtm516 committed:
[Android] Remove usage of MCProtocolLib Base64 in SkinUtils + more (#1237)

* Remove usage of MCProtocolLib Base64 in SkinUtils

* Fix path resolution for downloading locales

rtm516 committed:
Add support for manually supplying Bedrock resource packs (#1076)

* send resource packs

A lot of this code is nukkit-credits in the classes

* send resource packs

A lot of this code is nukkit-credits in the classes

* Remove unnecessary code/debugs

* use separately generated hashes

* Updated mappings and added .mcpack support

* "packs" directory auto-create (#484)

* "packs" directory auto-create

* cleaned indentation in ResourcePack.java

* Cleaned ResourcePack.java

* Another cleanup

I hate editor on github.

* Yet another

* Another indentation cleanup

* Fix resource pack loading

(cherry picked from commit f93b07491ea0d4cd96254af47a16f10ce9b92128)

* Move back to internal sha256 hashing

(cherry picked from commit 812a3d82b2a4d34e67bbccc193fbd87ac9fb174b)

* Add resource pack loading back after merge

* Add comments, config option and removed unused files

* Fix packs folder location and cleanup code

* Move to better options for the client

* Fix typos in comments

* Fix pack loading

* Try to make it compile

* Final touches?

* Add Javadoc for MathUtils#constrain

Co-authored-by: EOT3000 <43685885+EOT3000@users.noreply.github.com>
Co-authored-by: Vesek <61123478+Vesek@users.noreply.github.com>
Co-authored-by: Heath123 <heath.mitchell27@gmail.com>
Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Camotoy committed:
InventoryUtils: Don't send Java packet on hotbar item selection (#1301)

The Bedrock client sends a confirmation packet we translate regardless.

Comstepr committed:
Update pullrequest.yml (#1305)

Camotoy committed:
Only send the client brand on game join (#1299)

* Only send the client brand on game join

* Apply suggested changes

Camotoy committed:
Scoreboard: Fix various issues (#1286)

* Scoreboard: update score on UpdateType.ADD

* Actually fix

* Readd the Objective when a score changes

It looks like Objectives only update when you Remove the Objective and add it back using the SetDisplayObjective. This is hopefully a hotfix, but I think that there is no better way.

* Explain score tracking

Co-authored-by: Tim203 <mctim203@gmail.com>

Camotoy committed:
Remove warning about slot 50 (#1325)

We know that it occurs with console crafting.

Camotoy committed:
Implement experience sounds (#1320)

Bedrock sends a level event for the experience sound around the same time as a Java entity collect item packet is sent.

Redned235 committed:
Use SERVER_READY in BedrockRespawnTranslator to fix respawn bug

Camotoy committed:
Fix interaction spam bug (#1324)

* Fix interaction spam bug

This references the Nukkit 1.0 fix for the client bug of spamming to interact. Holding down still works.

* Remove interaction position set at action type 1

* Remove debug line

Camotoy committed:
Fix picking up liquids with buckets (#1311)

* Fix picking up liquids with buckets

The last fix to prevent bucket placement upon interacting with an inventory had an oversight with empty buckets, making them unusable. This commit fixes that while keeping the previous fix.

* Remove debug line

* Fix milk drinking and visual bucket item apperance

* Comment elaboration

* Make indentiation better

Camotoy committed:
Prevent Bedrock from changing gamemode client-side (#1337)

In vanilla Bedrock, if you have operator status, the client sends a packet to change gamemode without confirmation from the server. Since we have a custom server option to request the gamemode, we just reset the gamemode and ignore this packet.

lukeeey committed:
Update the CI link in the README (#1339)

rtm516 committed:
Fix gamemodes not fully applying on server switch (#1348)

* Fix gamemodes not fully applying on server switch

* Revert previous commit and move session flag updating to the adventure settings method

EasyClifton committed:
Update README.md (#1349)

Make the server address look cool (for build 405, sorry)

bundabrg committed:
Remove 'geyser' from parameters when executing a command under Spigot, Bungeecord, Sponge, Velocity (#1266)

* Remove 'geyser' from parameters when executing a command under Spigot, Bungeecode, Sponge, Velocity

Fixes https://github.com/bundabrg/GeyserReversion/issues/8

* Fix case when there are no sub commands

Co-authored-by: bundabrg <bundabrg@grieve.com.au>

bundabrg committed:
Merge branch 'master' into feature/custom-skulls-pr

# Conflicts:
#	bootstrap/sponge/src/main/java/org/geysermc/platform/sponge/GeyserSpongeConfiguration.java
#	connector/src/main/java/org/geysermc/connector/GeyserConnector.java
#	connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java
#	connector/src/main/java/org/geysermc/connector/entity/PlayerEntity.java
#	connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java
#	connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerListEntryTranslator.java
#	connector/src/main/java/org/geysermc/connector/network/translators/java/world/JavaChunkDataTranslator.java
#	connector/src/main/java/org/geysermc/connector/network/translators/java/world/JavaUpdateTileEntityTranslator.java
#	connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/SkullBlockEntityTranslator.java
#	connector/src/main/java/org/geysermc/connector/utils/ChunkUtils.java
#	connector/src/main/java/org/geysermc/connector/utils/SkinProvider.java
#	connector/src/main/java/org/geysermc/connector/utils/SkinUtils.java
#	connector/src/main/resources/config.yml

bundabrg committed:
Merge Fixes

Camotoy committed:
Villager trade fixes (#1350)

This commit mainly focuses on fixing the crashing of villagers that occurred pre-1.14.

Co-authored-by: AJ Ferguson <AJ-Ferguson@users.noreply.github.com>

Camotoy committed:
BedrockRespawnTranslator: prevent some respawn bugs (#1346)

Camotoy committed:
DoorSoundInteractionHandler: ignore iron [trap]doors (#1343)

Camotoy committed:
Tipped arrow translation (#1331)

* Tipped arrow translation

- Tipped arrow items are now properly translated both ways
- Tipped arrow particle effects are also translated, by having a list of all colors Java could send us and their Bedrock ID

* Remove a whitespace

Camotoy committed:
Forward keep alive packets to the client (#1344)

* Forward keep alive packets to the client

Previously, MCProtocolLib (our Java protocol library) handled keep alive packets for us. This commit disables that option and 'forwards' the keep alive packets to the client, and sending the keep alive packet back once Bedrock sends us a ping response.

* Delete DataCache

* Update to latest MCProtocolLib

* Swap values around as a sanity check

Camotoy committed:
Add support for fishing rods pulling Bedrock players (#1355)

Fishing rods pulling players is a clientside feature on Java. On Bedrock, a SetEntityMotionPacket is sent to the client. Therefore this PR implements the Java fishing rod pulling mechanics and sends it off to Bedrock, which sends MovePlayerPackets that are sent to the server.

Camotoy committed:
Jenkins improvements (#1368)

- Show changes in the Discord webhook
- Delete after 20 builds

Co-authored-by: rtm516 <ryantmilner@hotmail.co.uk>

Camotoy committed:
Add Fabric as a platform type (#1376)

* PlatformType: Add Fabric as a platform

* Don't use XML reflections on Fabric

Arktisfox committed:
Area cloud fixes (#684)

* Fix NotNull error with particles, replace incorrect string meta with int meta.

* Add back newline

* Remove debug line

* Update Protocol and prepare for merge

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
Fix fire being punched in all directions (#1370)

Apply the fix we have for fire but for all block faces.

Tim203 committed:
Various Scoreboard fixes (#1381)

* Various Scoreboard fixes

Fixes #1328 and a few other potential Scoreboard problems

* Consistent whitespacing

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
BedrockMobEquipmentTranslator: Don't change item slot if already on that slot (#1353)

* BedrockMobEquipmentTranslator: Don't change item slot if already on that slot

* Update comment

Camotoy committed:
Make crossbows prettier (#1359)

- Fix crossbow NBT translation - now crossbows will show as loaded
- Pillagers now more closely resemble Java Edition pose behavior

Camotoy committed:
Set default values for classes as well (#1387)

Geyser can now start even if the config file is empty. Tested on Spigot and doesn't affect custom values.

Camotoy committed:
Fix bucket interactions on creative mode (#1369)

* Fix bucket interactions on creative mode

Bedrock uses the BLOCK_INTERACT enum of BedrockActionTranslator to truly indicate if a bucket should be used or not. In order to hook into this, we need to delay the bucket placing by about 5 milliseconds - this gives us time to cancel the interaction if needed.

Bucket sounds will now not play in this case as well.

lukeeey committed:
Add player device OS to metrics (#1391)

* Add player device os to metrics

* Add player version, Geyser version, and default locale

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
Set AuthType in Metrics to lowercase (#1395)

Redned235 committed:
Fix area effect clouds

DaMatrix committed:
Don't use wrapper objects for positions in ChunkCache (#1398)

* make ChunkPosition use a hashCode implementation with far better hash distribution

this should improve the performance when used as a hash table key

* ChunkCache no longer uses position wrapper objects

this yields a roughly 15-20% increase in performance when converting chunk data

* fix code style issues

DaMatrix committed:
fix some NPEs caused by race conditions in chunk conversion (#1396)

* fix some NPEs caused by race conditions in chunk conversion

tbh the whole session should be read-write locked for every operation

* fix code style issues

OnlyBMan committed:
Merge pull request #9 from bundabrg/feature/custom-skulls-pr

Feature/custom skulls pr

Tim203 committed:
Don't use the general thread pool to run an async method (#1397)

* Don't use the general thread pool for an async method

* Align nested class at the bottom

Redned235 committed:
Add sound when an arrow hits a player

DaMatrix committed:
Faster chunk conversion (#1400)

* BlockStorage is never used concurrently, no need to synchronize

* initial, semi-functional, faster chunk conversion

* faster chunk conversion works well for every situation except spigot

* delete unused ChunkPosition class

* preallocate and pool chunk encoding buffers

* make it work correctly on spigot

* make field naming more consistent

* attempt to upgrade to latest MCProtocolLib

* remove debug code

* compile against my MCProtocolLib fork while i wait for my upstream PR to be accepted

* return to Steveice10 MCProtocolLib

Redned235 committed:
Fix wolf collar color when it's no longer angry (Closes #1404)

Camotoy committed:
JavaOpenWindowTranslator: Use MessageUtils for inventory name (#1416)

* JavaOpenWindowTranslator: Use MessageUtils for inventory name

* Remove important messaging

davchoo committed:
Fix mob mount positions (#1392)

* Fix mob mount positions

Uses offsets from Java Edition.

* Fix Boat mount pos for multiple passengers and Fix Ravager

Remove unnecessary horse metadata

* Fix Minecart & Boat Mount Pos, Fix Player Height Offset

* Use offset of EntityType.PLAYER

* Add back metadata

DaMatrix committed:
fix chunk section decoding (#1418)

* fix chunk section decoding

* switch back to official MCProtocolLib

Camotoy committed:
SettingsUtils: fix 'show coordinates' setting persistence (#1429)

The boolean that toggled this was accidentally in the wrong spot.

DaMatrix committed:
fix decoding sections with duplicate palette entries (#1430)

Camotoy committed:
GeyserSession: Set a default value for attackSpeed (#1419)

Fixes cooldowns not showing on a fresh world.

Camotoy committed:
Update to Adventure 4.1.1 (#1410)

* Update to Adventure 4.0.0

* Update to 4.0.1

* Update again, I guess.

Redned235 committed:
Make userAuths information more clear

rtm516 committed:
Bedrock <-> Bedrock skin display fix (#1195)

* Implement partial bedrock skin fix

* Fix equals method

* Fix ViaVersion

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Redned235 committed:
Translate emote list packet

Redned235 committed:
Use correct methods in refreshEmotes

rtm516 committed:
Add statistics menu (#1424)

* Add statistics menu

* Changed back button text

* Add check to make sure the player requested the statistics display

* Better item translation support; misc changes

* Clean up session getting?

* Remove extra debug that is likely unnecessary

* Remove unused function

* Update languages submodule

* Clean up javadoc comment

* Fix typo

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>
Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

davchoo committed:
Projectile fixes (#1451)

* Predict the trajectory of projectiles and add particles

* Correct lingering potion gravity

* Update last position on move absolute

* Clean up

* Add egg to ItemRegistry and update mappings

SupremeMortal committed:
Use new Open Collaboration maven repository

Camotoy committed:
GUI Improvements (#1462)

- Added `GeyserCommand.isExecutableOnConsole()`. If this is set to false, the command will not appear as an option in the GUI.
- Added `GeyserCommand.getSubCommands()`. If not empty, the subcommand options will now appear in the GUI.

Camotoy committed:
BedrockActionTranslator: Fix occasional death stall (#1432)

Usually this happened when joining from another dimension after the player exited to the main menu on the death screen. The player would not realize that they are dead.

Camotoy committed:
Introduce CommandSender.getLocale() (#1431)

* Introduce CommandSender.getLocale()

This allows Geyser-specific commands (e.g. `/geyser help`) to be displayed in the (Java or Bedrock) player's default language, which stops those commands from simply being displayed in the default locale.

* Tweak Javadoc

* Set CommandManager's GeyserConnector to final

* Clean up

davchoo committed:
Ender dragon Melee Attacks (#1466)

* Create and position Ender Dragon Bounding Box

Currently allows the player to "kill aura" target
the ender dragon.

* Use an entity to handle attacks for each hitbox

* Use the proper flag to make entities invisible

* Clean up and add some comments

* Ender dragon entity metadata improvements

* Add doc to segment functions

* Add changes

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
Add support for more recipes (#1434)

* Add support for more recipes

- Tool repairing
- Book cloning
- Suspicious stew
- Tipped arrows

What still needs to be done:

- Map cloning/extending (though there may be item mapping mismatch getting in the way)
- Banner duplication (couldn't figure this out)

* Add some more spacing

* Explain recipe UUIDs

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into custom-skulls

Camotoy committed:
MapItemTranslator: support map tag being of short value (#1475)

GSigns sends this as a short. Who knows why.

Camotoy committed:
Refactor skulls a bit

Camotoy committed:
Update to 1.16.4 (#1487)

* 1.16.4-pre1 support

* Update to support ViaVersion 3.2.0 (#1358)

* Update to support ViaVersion 3.2.0

This commit updates ViaVersion integration to support their recent mappings changes. 

Co-authored-by: Nassim <jahnke.nassim@gmail.com>

* Send adventure settings on gamemode change after gamemode swap

* Update Velocity to 1.1.0 proper

* Update to 1.16.4

Co-authored-by: Nassim <jahnke.nassim@gmail.com>

Camotoy committed:
Send position update every 3 seconds if idle (#1421)

* Send position update every 3 seconds if idle

Prevents timeouts in certain instances when AFK.

* Cancel position sending on dimension switching

* Remove debug lines

* Create function to centralize movement translation

matthi4s committed:
Update supported version in README to 1.16.4 (#1488)

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into custom-skulls

Camotoy committed:
Further optimizations

davchoo committed:
Fix sitting with armor stand and animal mount offsets (#1492)

* Fix sitting with armor stand and animal offsets

* Fix riding players

* Moved @Getter

Camotoy committed:
Fix bell sound and visuals (#1502)

* Fix bell sound and visuals

The bell sound now correctly plays.

Bells will also visually ring when rung by another player.

* Compress elses

* Add more whitespace to new code

Tim203 committed:
Block entity performance improvements (#1481)

* BlockEntity performance improvements

* Use chunk cache if possible for block caching

* Get new block state from ViaVersion if block entity

* Add Javadoc for FlowerPotBlockEntityTranslator.isFlowerBlock

* Remove debug line

* Don't add all RequiresBlockState instances if cache chunks is enabled

* Double chest map get optimization

* Last changes

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
Fireball and ghast improvements (#1469)

* Fireball and ghast improvements

- Ghasts now visually show if they're charging a fireball
- Fireballs are now vastly better and will update better

* Add gravity and drag to projectiles

* Add check for session close and improve fireball

* Remove motion stuff from fireball

* Make fireball hittable

* Add wither skull entity

* Small changes

* Add note about laggy fireballs

Co-authored-by: David Choo <davchoo3@gmail.com>

rtm516 committed:
Add config option for enabling achievements (#1504)

* Add config option for enabling achievements

* Disabled achievements by default and added warning about commands being disabled

* Update config.yml

* Rename achievements-enabled to xbox-achievements-enabled for clarity

Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

rtm516 committed:
Disable the sponsor button (#1509)

rtm516 committed:
Add support for passing config options as arguments (#1506)

Tim203 committed:
Fixed some issues related to Scoreboards (#1446)

* Fixes some issues related to Scoreboard Teams

* The cached Score info should update no matter what kind of update the Team got.
* Team entities specified at the create Team packet should also be checked if they exist as Score in the registered Objectives

* Rewrote some Scoreboard code and fixed various issues

* Minor formatting changes

Camotoy committed:
Update MCProtocolLib to fix datapacks that send empty result recipes (#1522)

Example: https://www.planetminecraft.com/data-pack/true-survival-a-hardcore-minecraft-experience/

davchoo committed:
Fix laying crash (#1510)

* Fix crash with GSit lay and use Java bed position data

* Fix GSit's lay position

* Move Bed Position metadata to the right class

* Actually fix lay for PosePlugin

* Revert "Actually fix lay for PosePlugin"

This reverts commit 3f21261162c439b3b167f057e75c2884193b75f7.

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

rtm516 committed:
Fix even more entity metadata flags (#1483)

* Fix even more entity metadata flags

* Add comment explaining magic value

* Fix horse flags and add more information

* Add more information about the Horse eating particles

Camotoy committed:
Add more interactive tags (mobile buttons) (#1443)

* Add more interactive tags (mobile buttons)

This expands our support for showing the interactive tags on touchscreen and console setups. This is not complete - specifically, the food compatibility of creatures needs to be expanded upon (I will work on this later and does not stop this PR from being mergable). This also includes:

- Creepers who are ignited with flint and steel now show up properly
- Zombie villagers now shake properly when converting and show their region outfits

* Add more food choices and add more panda entity metadata

* Re-add eating flag

* Remove debug line

* Refactor dimension usage, finish interactive tag usage, bees

* Print statements... ._.

* Don't make eating item packet data a non-constant

* Move BAMBOO to ItemRegistry

* Add missing break

* Make changes

* Minor final changes

Camotoy committed:
Update MCProtocolLib to fix more custom recipe stuff (#1534)

niklaswa committed:
Update LabyMod cape url (#1540)

Camotoy committed:
Fix GlobalPalette translation (#1528)

* Fix GlobalPalette translation

Global palettes don't have their own internal palette, which cannot be iterated through to create a Bedrock palette. Therefore we simply iterate over the whole palette one time. This commit also fixes a regression with flowers/pistons being on multiple chunk sections.

* Don't declare bedrockPalette until after global palette check

Redned235 committed:
Move PlatformType to common module

Camotoy committed:
BannerTranslator: fix NPE when no block entity data (#1543)

rtm516 committed:
Rewrite message handling in MessageUtils to use Adventure (#1498)

* Rewrite message handling in MessageUtils to use Adventure

* Move to static Adventure commit to fix a bug

* Initial test implementation

* Add RGB downgrade test

* Move MessageUtils and rename

* Clean-up and fix tests

* Fixed sign and book content handling

* Fix blank signs causing NPEs

* Fix reset before message being stripped

* Add comment about the reset character

* Fix legacy style server motds

* Fix more messages being handled wrong

* Fix title packets being handled wrong

* Fix trailing formatting characters on the end of sign lines

* Add auto updating of Java locale files

* Add en_us locale updating and hash caching

* Changes to hash determining

Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>

Camotoy committed:
Update to Bedrock 1.16.100 (#1552)

* Initial work on 1.16.100 - currently crashes the client

* Update runtime item states

* Use new Bedrock runtime IDs

Bedrock now hardcodes block runtime IDs in alphabetical order of the identifiers. This commit updates Geyser to accomodate.

- Remove runtime_block_states.dat and replace it with blockpalette.nbt
- Calculate the block runtime ID based on the order of the block palette
- Separate BlockTranslator.AIR into Bedrock and Java values
- Update the second layer of chunks to use air when not waterlogged
- Don't send item palette for now, as that's what crashes the game (will look into for v415)
- Other misc. changes

* Improve second layer chunk translation

* v415 support

- Add a message warning people they are on a beta version of Geyser

* Update to protocol v417

There are still some mappings changes that need to be gone through.

* Update runtime item states and clean up item frames

* Future-proof enchanment table

* Update for v418

* Update to v419

* Apply proper air ID to waterlogged chunk layer

* Fix missing import

* Remove beta warning

* Update mappings

* Manually patch runtime_item_states and send the ITEMS registry

* Update README

* Disable grindstone and smithing inventories (since they're broken)

* Use artifactory jenkins plugin (#1548)

* Use artifactory jenkins plugin

* Bump version to 1.2.0-SNAPSHOT

Co-authored-by: SupremeMortal <6178101+SupremeMortal@users.noreply.github.com>

Camotoy committed:
Revert artifactory changes preventing compilation (#1555)

Camotoy committed:
Print reason for disconnect when outdated (#1556)

Camotoy committed:
Fix regressions from 1.16.100 (#1558)

* Fix regressions from 1.16.100

- Update mappings to fix recipe regressions and item differences
- Villager trading NBT now prefers the String identifier and not the integer ID

* Fix lodestone compass breaking

rtm516 committed:
More chat fixes (#1557)

* Fix positional translation arguments not being handled

* Fix locale fallback

* Fix command completion

* Remove the expensive call to `containsKey`

* Unify adventure versions

* Fix some more formatting issues due to parity

* Fix and update tests

* Update adventure

* Add Javadoc for getCommandNames

* Formatting

Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Heath123 committed:
Fix inconsistencies with movement and position (#699)

Movement is now significant better, especially on slabs, stairs, and other half-blocks.

Co-authored-by: RednedEpic <redned235@gmail.com>
Co-authored-by: DoctorMacc <toy.fighter1@gmail.com>
Co-authored-by: Tim203 <mctim203@gmail.com>
Co-authored-by: Camotoy <20743703+DoctorMacc@users.noreply.github.com>

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into custom-skulls

Camotoy committed:
Fix weird skin behavior when player head and player is present

Camotoy committed:
LivingEntity: add visual support for riptide spin attack (#1551)

OnlyBMan committed:
Merge pull request #10 from DoctorMacc/custom-skulls

Update to 1.16.100 and refactor

rtm516 committed:
Clean-up translation string implementation (#1567)

Camotoy committed:
Set own player profile earlier on

Camotoy committed:
Fix inconsistencies when using WorldEdit

Camotoy committed:
Merge branch 'custom-skulls' of https://github.com/OnlyBMan/Geyser into custom-skulls

rtm516 committed:
Fix en_us downloading and hashing on Android (#1574)

rtm516 committed:
Another android fix (#1575)

* Fix en_us hash reading on Android (again)

* Fix hash generation methods for Android

rtm516 committed:
Clear the Reflections scanners to stop it trying to scan (#1582)

Camotoy committed:
More polish

Redned235 committed:
Add support for latest beta (1.16.200.56)

Carbuino committed:
Update UpstreamPacketHandler.java (#1570)

Changed the logging in message to display in the Action Bar instead of spamming the chat.

bundabrg committed:
Ignore missing quotes in YML files

Closes #10

Camotoy committed:
Fix mounts being unmountable with cache chunks (#1576)

Teleports need to be confirmed before riding

rtm516 committed:
Update MCProtocolLib to Adventure migration (#1572)

* Use raw message data instead of converting the message

* Update MCProtocolLib to Adventure

* Ignore MCProtocolLib Adventure depend

* Remove unused dependency

* Fix isMessage handling `null` wrong.

* Update to adventure 4.2.0

* Clean-up isMessage

* Fix tests

* Clean-up of catch statements

Camotoy committed:
GUI: Don't exit if there is a config error (#1600)

Leave the window open so the user understands there is an error.

This also changes the exit code to 1 when exiting without a GUI since we have an error.

Camotoy committed:
Prepare for more refactors

Camotoy committed:
Add separate skull skin manager

rtm516 committed:
Add check to fix NONE team color causing an NPE (#1602)

Camotoy committed:
A bunch of stuff

- Fix reloads changing the skull config
- If a skull doesn't have a properties section, download it
- Add item translation for custom skulls

Camotoy committed:
Tiny changes

Camotoy committed:
Fix villagers accepting books (#1605)

* Hopefully fix villagers accepting books

Fixes a couple of inconsistencies getting books from the creative menu.

* Fix fake news

Camotoy committed:
Add requested changes and fix possible NPE

Camotoy committed:
Make JSON method in WebUtils

Bastian committed:
Do not send server custom chart for Metrics (#1610)

The server chart is automatically populated by the bStats backend.

Camotoy committed:
Hide invalid username errors

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into custom-skulls

Camotoy committed:
Supply a unique network ID for each recipe (#1615)

This fixes crashes in the Minecraft betas.

basaigh committed:
Bump languages submodule (#1619)

rtm516 committed:
Fix supported version display name (#1593)

* Fix supported version display name

* Update connector/src/main/java/org/geysermc/connector/network/BedrockProtocol.java

Camotoy committed:
Merge branch 'master' of https://github.com/GeyserMC/Geyser into custom-skulls

Camotoy committed:
Clean up and remove schedule entirely

Since we aren't using it yet anyway.

Camotoy committed:
Fix documentation

Camotoy committed:
Fix wrong method being called

Camotoy committed:
Use get instead of getOrDefault

Camotoy committed:
Update config to set default value
